lib/cmake.rb

```diff
@@ -151,34 +151,38 @@ def cmake_package(compiler, src_dir, build_dir, build_type)
       # :nocov:
     end
 
-    if !compiler[:package_command].nil?
-      pack_stdout, pack_stderr, pack_result = run_scripts(
-        @config,
-        ["cd #{build_dir} && #{compiler[:package_command]} "], 'PATH' => new_path
-      )
-    else
-      pack_stdout, pack_stderr, pack_result = run_scripts(
+    # initialize some variables to accrue over all generators
+    pack_stdout, pack_stderr = '', ''
+    pack_result = 0
+
+    # convert the generator to an array if only one was given
+    generators_to_use = compiler[:build_package_generator]
+    generators_to_use = [generators_to_use] unless generators_to_use.kind_of?(Array)
```

Comment:
This line allows the user to specify a scalar generator in the config, they don't _have_ to use a list.  But in here we will construct a list either way.